src/v/cloud_storage/partition_manifest.cc

```diff
@@ -1279,6 +1280,7 @@ void partition_manifest::serialize(std::ostream& out) const {
         w.StartObject();
         for (const auto& [key, meta] : _segments) {
             serialize_meta(meta);
+            co_await ss::coroutine::maybe_yield();
```

Comment:
If `serialize_meta` is also adjusted to return a future, then this loop can be distributed to all shards.

Reply:
Not sure I'm following -- is the idea to try and share the serialization work across shards?  That would quite a big change